#### This is Syed Hamza from BDAT 1004 24F. This is the <mark> Question No. 21 </mark> of my Mid Term Exam

In [53]:
class Point:  #creating the Point class
    def __init__(self, x=0, y=0):        #__int()__: Constructor that takes as input a pair of Point objects that represent the ends points of the line segment
        'initialize point pairs'
        self.x = x
        self.y = y
    
    def get(self):
        return(self.x, self.y)
    
    def __repr__(self):
        return 'Point({},{})'.format(self.x,self.y)
    
    def __add__(self, point):
        return Point(self.x+point.x, self.y+point.y)    
    
    def __sub__(self, point):
        return Point(self.x-point.x, self.y-point.y)
    
class Segment:
    def __init__(self, p1=Point(0,0), p2=Point(0,0)):
        self.p1 = p1
        self.p2 = p2
    
    def get(self):
        return(self.p1, self.p2)
    
    def __repr__(self):
        return 'Segment({},{})'.format(self.p1, self.p2)
    
    def length(self):  #Length():: returns the length of the segment 
        return (((self.p1-self.p2).x)**2 + ((self.p1-self.p2).y)**2)**(1/2)
    
    def slope(self):  #Slope() returns the slope of the segment of none if the slope is unbounded 
        if self.p1.x == self.p2.x:
            if self.p1.y == self.p2.y:
                return("None")
            else:
                return("Infinity")
        else:
            return (self.p1-self.p2).y / (self.p1-self.p2).x

In [55]:
#checking the given code in the exam to generate the expected output

p1 = Point(3,4)

p2 = Point()

s = Segment(p1,p2)

print(s.length())

print(s.slope())    #for this question, the output is shown as 0.75 in the exam question, I think there could be an error  with the formula applied

5.0
1.3333333333333333


#### Reflection:

I think there was a mistake with the execution of s.slope() as the answer turns out to be 1.33 compared to the expected 0.75. The answer will be only 0.75 for s.slope() if we create p1 as Point(4,3).